homeassistant/components/mqtt/image.py

```diff
@@ -0,0 +1,199 @@
+    """representation of a MQTT image."""
+
+    _entity_id_format: str = image.ENTITY_ID_FORMAT
+    _attributes_extra_blocked: frozenset[str] = MQTT_IMAGE_ATTRIBUTES_BLOCKED
+    _last_image: bytes | None = None
+    _client: httpx.AsyncClient
+    _template: Callable[[ReceivePayloadType], ReceivePayloadType]
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        config: ConfigType,
+        config_entry: ConfigEntry,
+        discovery_data: DiscoveryInfoType | None,
+    ) -> None:
+        """Initialize the MQTT Image."""
+        self._client = get_async_client(hass)
+        ImageEntity.__init__(self)
+        MqttEntity.__init__(self, hass, config, config_entry, discovery_data)
+
+    @staticmethod
+    def config_schema() -> vol.Schema:
+        """Return the config schema."""
+        return DISCOVERY_SCHEMA
+
+    def _setup_from_config(self, config: ConfigType) -> None:
+        """(Re)Setup the entity."""
+        if not config[CONF_FROM_URL]:
+            self._attr_content_type = config[CONF_CONTENT_TYPE]
+        self._template = MqttValueTemplate(
+            config.get(CONF_VALUE_TEMPLATE), entity=self
+        ).async_render_with_possible_json_value
+
+    async def _async_load_image(self, url: str) -> None:
+        try:
+            response = await self._client.request("GET", url)
+        except (httpx.TimeoutException, httpx.RequestError, ssl.SSLError) as ex:
+            _LOGGER.warning("Connection failed to url %s files: %s", url, ex)
+            return
+
+        try:
+            content_type = validate_content_type(response.headers["content-type"])
+        except vol.Invalid as err:
+            _LOGGER.error(
+                "Content is not a valid image, url: %s, content_type: %s, %s",
+                url,
+                response.headers["content-type"],
+                err,
+            )
+            return
+        self._attr_content_type = content_type
+        self._last_image = response.content
+        self._attr_image_last_updated = dt_util.utcnow()
+        self.async_write_ha_state()
+
+    def _prepare_subscribe_topics(self) -> None:
+        """(Re)Subscribe to topics."""
+
+        @callback
+        @log_messages(self.hass, self.entity_id)
+        def message_received(msg: ReceiveMessage) -> None:
+            """Handle new MQTT messages."""
+
+            if self._config.get(CONF_FROM_URL, True):
+                try:
+                    url = cv.url(self._template(msg.payload))
+                except vol.Invalid:
+                    _LOGGER.error(
+                        "Invalid image URL '%s' received at topic %s",
+                        msg.payload,
+                        msg.topic,
+                    )
+                    return
+                self.hass.async_create_task(self._async_load_image(url))
+                return
+            if CONF_IMAGE_ENCODING in self._config:
+                self._last_image = b64decode(msg.payload)
```

Comment:
If this fails, should we set the image to None?

Reply:
As far as I know we do not update the state in case of an error in with other MQTT platforms. But may be could catch exceptions here and log a warning.

Reply:
I have updated the code to set the image to None